src/modules/local_position_estimator/BlockLocalPositionEstimator.cpp

```diff
@@ -850,6 +852,31 @@ void BlockLocalPositionEstimator::predict()
 	_P += (_A * _P + _P * _A.transpose() +
 	       _B * _R * _B.transpose() +
 	       _Q) * getDt();
+
+	// condition P
+	for (int i = 0; i < n_x; i++) {
+		for (int j = 0; j <= i; j++) {
+			if (_P(i, j) > P_MAX) {
+				_P(i, j) = P_MAX;
+
+			} else if (!PX4_ISFINITE(_P(i, j))) {
+				_P(i, j) = 0;
```

Comment:
Is it safe to write 0 in the diagonal of P? This is saying that if the covariance of state i becomes infinite, reset it to 100% certainty.